src/main/java/org/gridsuite/securityanalysis/server/dto/ContingencyResultDTO.java

```diff
@@ -36,4 +41,19 @@ public static ContingencyResultDTO toDto(ContingencyEntity contingency) {
             .subjectLimitViolations(subjectLimitViolations)
             .build();
     }
+
+    // each contingencyResultDto will return multiple line (one for each limitViolation)
+    public List<List<String>> toCsvRows(Map<String, String> translations) {
+        List<List<String>> csvRows = new ArrayList<>();
+        this.getSubjectLimitViolations().forEach(lm -> {
+            List<String> csvRow = new ArrayList<>();
```

Comment:
Optimize array memory pre-allocation
```suggestion
            List<String> csvRow = new ArrayList<>(3);
```